docs/user/openstack/affinity.md

```diff
@@ -0,0 +1,136 @@
+# How to set affinity rules for workers at install-time
+
+The Installer automatically creates Control plane nodes with a
+"soft-anti-affinity" policy. This means that each server will be created on a
+separate host, if enough hosts are available.
+
+The Installer does not set any affinity policy on Compute nodes; the choice is
+left to the user.
+
+We are going to create an OpenStack `server group` with an `anti-affinity`
+policy. Then we are going to modify the `MachineSet` definitions during the
+installation process, so that when `openshift-install create cluster` is run,
+the Compute nodes will directly be created with the given policy.
+
+### Step 1: create the Server group
+
+In OpenStack Compute (Nova), scheduler policies are set using Server groups.
+
+The first step is then to create the Server group.
+
+Depending on the OpenStack configuration, there are a maximum of four possible policies:
+* affinity
+* soft-affinity
+* anti-affinity
+* soft-anti-affinity
+
+These options are best described in the [OpenStack API reference][openstack-compute-api-docs].
+
+Here we set `anti-affinity`:
+
+```shell
+openstack \
+	--os-compute-api-version=2.15 \
+	server group create \
+	--policy anti-affinity \
+	my-openshift-worker-group
+```
+
+Note that setting the Nova microversion ("os-compute-api-version") is only
+mandatory when dealing with the "soft" policies, but does not hurt in any case.
+
+**Take note of the resulting Server group id.**
+
+### Step 2: generate the OpenShift manifests
+
+```shell
+openshift-install create manifests
+```
+
+This command will interactively prompt for the required information, if an
+`install-config.yaml` file is not present in the current working directory.
+
+### Step 3: Edit the MachineSet manifests
+
+The "create manifests" command creates several files, among which the worker
+MachineSet definition in
+`openshift/99_openshift-cluster-api_worker-machineset-0.yaml`.
+
+We must add a new property to the MachineSet: `serverGroupID`. That property lives under `spec.template.spec.providerSpec.value`.
+
+**Use the Server group id from step 1.**
+
+I have used the example UUID `aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee` in the template below.
```

Comment:
Stylistic note: you've also used <infrastructure_ID>, <node_role>, <nova_flavor>, etc. You should probably do with with <server_group_uuid>, and perhaps call it out here because it's important.